scanners/headless/base.js

```diff
@@ -0,0 +1,25 @@
+'use strict';
+
+
+var scan = async function (domain, environment, options, getBrowser, scanner, callback) {
+
+  const browser = await getBrowser();
+
+  const page = await browser.newPage();
+  var data;
+
+  // Do the scanner-specific heavy lifting.
+  try {
+    data = await scanner.scan(domain, environment, options, browser, page);
+  } catch (exc) {
+    await browser.close();
+    return callback(exc)
+  }
+
+  await browser.close();
+
+  // TODO: error handling
+  return callback(null, data);
+};
+
+module.exports = {scan: scan}
```

Comment:
This `base.js` file doesn't do a whole lot, since `local_bridge.js` and `handler.js` handle a lot of the setup work before executing a specific scanner. But it does do some stuff, that specific scanners won't have to do, and I suspect this will grow to support a bunch of common kinds of error handling and options management.